falco.yaml

```diff
@@ -1206,33 +1206,44 @@ base_syscalls:
-# namespace and pod name directly from the container runtime, see
-# https://falco.org/docs/reference/rules/supported-fields/#field-class-container).
-#
-# Furthermore, Falco exposes container events themselves as a data source for
-# alerting. To achieve this integration with the container runtime, Falco
-# requires access to the runtime socket. By default, for Kubernetes, Falco
-# attempts to connect to the following sockets:
-# "/run/containerd/containerd.sock", "/run/crio/crio.sock",
-# "/run/k3s/containerd/containerd.sock". If you have a custom path, you can use
-# the `--cri` option to specify the correct location.
-#
-# In some cases, you may encounter empty fields for container metadata. To
-# address this, you can explore the `--disable-cri-async` option, which disables
-# asynchronous fetching if the fetch operation is not completing quickly enough.
-#
-# To get more information on these command-line arguments, you can run `falco
-# --help` in your terminal to view their current descriptions.
-#
-# !!! The options mentioned here are not available in the falco.yaml
-# configuration file. Instead, they can can be used as a command-line argument
-# when running the Falco command.
+# [Incubating] `container_engines`
+#
+# This option allows you to explicitly enable or disable API lookups against container 
+# runtime sockets for each supported container runtime, tracked internally as `container_engines`. 
+# Access to these sockets enables Falco to provide container and Kubernetes fields, which 
+# are crucial for identifying workload owners in modern containerized environments. 
+# Refer to the fields docs:
+# 
+# - [Kubernetes fields](https://falco.org/docs/reference/rules/supported-fields/#field-class-k8s)
+# - [Container fields](https://falco.org/docs/reference/rules/supported-fields/#container)
+# 
+# Additionally, Falco uses container events as a data source for alerting.
+# 
+# For most container engines, you can enable or disable them, and Falco will search the 
+# default container runtime socket paths, such as `/var/run/docker.sock` for Docker. 
+# However, for Kubernetes settings, you can customize the CRI socket paths:
+# 
+# - `container_engines.cri.cri`: Pass a list of container runtime sockets.
+# - `container_engines.cri.disable-cri-async`: Since API lookups may not always be quick or 
+# perfect, resulting in empty fields for container metadata, you can use this option option 
+# to disable asynchronous fetching. Note that missing fields may still occasionally occur.
+# 
+# The equivalent (stable) CLI args are `--cri` or `--disable-cri-async`.
+
+container_engines:
+  docker:
+    enabled: true
+  cri:
+    enabled: true
+    cri: ["/run/containerd/containerd.sock", "/run/crio/crio.sock", "/run/k3s/containerd/containerd.sock"]
+    disable-cri-async: false
+  podman:
+    enabled: true
+  lxc:
+    enabled: true
+  libvirt_lxc:
+    enabled: true
+  rocket:
+    enabled: true
```

Comment:
AFAIK `rocket` was the initial name, but its official name is `rkt`

Still, `rkt` was discontinued in 2020 (https://github.com/rkt/rkt), so I don't see any compelling reason to keep it.

@falcosecurity/libs-maintainers wdyt? :thinking:

Reply:
Oh I didn't know, yes can change it to `rkt`.

Reply:
> @falcosecurity/libs-maintainers wdyt? 🤔

Agree, we can clean it up; +1 from me!

Reply:
Removed it. It's not relevant for Falco as you pointed out.